xc/common/utils/prjxray_create_place_constraints.py

```diff
@@ -322,9 +333,18 @@ def __init__(self, vpr_grid, io_locs, blif_data, roi):
 
                     clock['sink_nets'].append(sink_net)
 
-                    assert sink_net in self.input_pins or sink_net in self.clock_sources, (
-                        sink_net, self.input_pins, self.clock_sources.keys()
-                    )
+                    if sink_net not in self.input_pins and sink_net not in self.clock_sources:
+
+                        # Allow BUFGs to be driven by generic sources but only
+                        # when enabled.
+                        if bel == "BUFGCTRL_VPR" and allow_bufg_logic_sources:
+                            continue
+
+                        # The clock source comes from logic, disallow that
+                        eprint(
```

Comment:
This is not actually an error, and it doesn't disallow it?

Reply:
I added an exit statement below.

When we talked about it we wanted to generally disallow logic driving clock sinks. There is an exception though for a `BUFGCTRL_VPR` but only enabled when the flag `allow_bufg_logic_sources` is set.

Reply:
There is still no exit?

Reply:
Unclear what happened before but the exit statement should be there now.